### PR TITLE
Handle the error when API id send as a UUID

### PR DESCRIPTION
**Pull request by shanakama**
## Purpose
Fix https://github.com/wso2-enterprise/choreo/issues/11694

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 1]
```diff
@@ -3263,6 +3263,7 @@ public void deleteAPI(String apiUuid, String organization) throws APIManagementE
                     "the recommendation system. "
             );
             isError = true;
+            throw e;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:3266]: Do we have a requirement to throw this?
The flow we have designed here is to try all the delete steps despite of the failures. ( eg: 
```
try { 
    some delete step()
} catch (failure) {
    log()
}

try { 
    next delete step()
} catch (failure) {
    log()
}

etc
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 2]
```diff
@@ -7895,13 +7896,18 @@ public API getAPIbyUUID(String uuid, String organization) throws APIManagementEx
...
+                throw new APIMgtResourceNotFoundException(e);
+            }
             throw new APIManagementException("Failed to get API", e);
         } catch (OASPersistenceException e) {
             throw new APIManagementException("Error while retrieving the OAS definition", e);
         } catch (ParseException e) {
             throw new APIManagementException("Error while parsing the OAS definition", e);
         } catch (AsyncSpecPersistenceException e) {
             throw new APIManagementException("Error while retrieving the Async API definition", e);
+        } catch (IllegalArgumentException e) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:7909]: Do we not send any APIPersistenceException from the persistent layer?.. If we do, do we have a requirement to catch a IllegalArgumentException?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 1 above)
Comment (shanakama) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:3266]: If it's unable to find the relevant API  , do we need to execute next steps ?? can't we ignore the next executions and throw the error ?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 2 above)
Comment (shanakama) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Removed via : https://github.com/wso2/carbon-apimgt/pull/11114/commits/e942ee49f9801090a2fc64e0dcd033619fee2eb9

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 1 above)
Comment (shanakama) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Fixed via : https://github.com/wso2/carbon-apimgt/pull/11114/commits/e41d31f1ee471c3d908a363cc5493708e6f46f26

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**